internal/memdb/memdb.go

```diff
@@ -0,0 +1,201 @@
+package memdb
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/cloudquery/plugin-sdk/plugins/destination"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+// client is mostly used for testing the destination plugin.
+type client struct {
+	schema.DefaultTransformer
+	spec          specs.Destination
+	memoryDB      map[string][][]any
+	errOnWrite    bool
+	blockingWrite bool
+}
+
+type Option func(*client)
+
+func WithErrOnWrite() Option {
```

Comment:
Intended for testing, these options should have `Test` somewhere (either godoc or in the option name, like `WithTestingErrOnWrite`, `WithTestingBlockingWrite`)

Reply:
those are all under `internal` so it can't be exported anyway for outside use.